src/main/java/backend/stub/DummyRepoState.java

```diff
@@ -400,6 +398,41 @@ protected TurboUser deleteUser(String idString) {
         return newLabels.stream().map(new Label()::setName).collect(Collectors.toList());
     }
 
+    protected final Issue setMilestone(int issueId, Optional<Integer> milestone) {
+        ImmutablePair<TurboIssue, IssueMetadata> mutables = produceMutables(issueId);
+        TurboIssue issueToSet = mutables.getLeft();
+        IssueMetadata metadataOfIssue = mutables.getRight();
+        List<TurboIssueEvent> eventsOfIssue = metadataOfIssue.getEvents();
+
+        // demilestone the issue, then set issue milestone using the new milestone
+        issueToSet.getMilestone()
+                .ifPresent(issueMilestone -> demilestoneIssue(issueMilestone, issueToSet, eventsOfIssue));
+        milestone
+                .ifPresent(newMilestone -> milestoneIssue(newMilestone, issueToSet, eventsOfIssue));
+
+        issueToSet.setUpdatedAt(LocalDateTime.now());
+        // Replace originals with copies, and queue them up to be retrieved
+        markUpdatedEvents(issueToSet, IssueMetadata.intermediate(eventsOfIssue, metadataOfIssue.getComments(), "", ""));
+
+        Issue serverIssue = new Issue();
+        milestone.ifPresent(newMilestone -> serverIssue.setMilestone(new Milestone().setNumber(newMilestone)));
+        return serverIssue;
+    }
+
+    private void milestoneIssue(int milestone, TurboIssue toSet, List<TurboIssueEvent> eventsOfIssue) {
+        eventsOfIssue.add(new TurboIssueEvent(new User().setLogin("test"),
+                IssueEventType.Milestoned,
+                new Date()).setMilestoneTitle(milestones.get(milestone).getTitle()));
+        toSet.setMilestoneById(milestone);
+    }
+
+    private void demilestoneIssue(int milestoneOfIssue, TurboIssue toSet, List<TurboIssueEvent> eventsOfIssue) {
```

Comment:
Should we use `schedule` and `unschedule` instead of `milestone` and `demilestone`? In any case add header comments to these two methods.